crates/oxc_linter/src/lint_adapter.rs

```diff
@@ -0,0 +1,1208 @@
+        }
+    };
+}
+
+macro_rules! once_vertex_iter_if_some {
+    ($maybe_some:expr, $vertex_type:ident) => {
+        if let Some(some) = &$maybe_some {
+            Box::new(std::iter::once(Vertex::$vertex_type(&some.0)))
+        } else {
+            Box::new(std::iter::empty())
+        }
+    };
+}
+
+impl<'a> LintAdapter<'a> {
+    fn get_ast_node_iter(&self) -> VertexIterator<'a, Vertex<'a>> {
+        Box::new(self.semantic.nodes().nodes.clone().into_iter().map(|v| Vertex::AstNode(v)))
+    }
+}
+
+fn solve_member_expr_for_lowest<'a>(object: &'a Expression<'a>) -> &'a Expression<'a> {
+    let mut obj = object;
+
+    while let Expression::MemberExpression(b) = obj {
+        obj = b.object();
+    }
+
+    obj
+}
+
+fn formal_parameters_to_iter<'a>(
+    params: &'a FormalParameters<'a>,
+) -> Box<dyn Iterator<Item = Vertex<'a>> + 'a> {
+    Box::new(
+        params
+            .items
+            .iter()
+            .enumerate()
+            .map(|(index, item)| Vertex::Parameter(&item.pattern.kind, index.try_into().unwrap())),
+    )
+}
+
+fn field_value_from_template_lit(tlit: &TemplateLiteral) -> FieldValue {
+    if tlit.expressions.len() == 0 && tlit.quasis.len() == 1 {
+        let quasi = &tlit.quasis[0].value;
+        FieldValue::String(quasi.cooked.as_ref().unwrap_or_else(|| &quasi.raw).as_str().into())
+    } else {
+        FieldValue::Null
+    }
+}
+
+fn assignment_target_to_vertex<'a>(at: &'a AssignmentTarget<'a>) -> Vertex<'a> {
+    match &at {
+        AssignmentTarget::SimpleAssignmentTarget(simple_assignment_target) => {
+            match simple_assignment_target {
+                SimpleAssignmentTarget::AssignmentTargetIdentifier(ident) => {
+                    Vertex::AssignmentToIdentifier(&ident.span, &ident.name)
+                }
+                _ => Vertex::AssignmentOnObject(&simple_assignment_target),
+            }
+        }
+        AssignmentTarget::AssignmentTargetPattern(d) => {
+            Vertex::DestructuringAssignment(DestructuringAssignment::AssignmentExpression(d))
+        }
+    }
+}
+
+// impl From<&AssignmentTargetPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &AssignmentTargetPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::AssignmentExpression(ap))
+//     }
+// }
+
+// impl From<&ArrayPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &ArrayPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::ArrayPattern(ap))
+//     }
+// }
+
+// impl From<&ObjectPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &ObjectPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::ObjectPattern(ap))
+//     }
+// }
+
+// impl From<&AssignmentPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &AssignmentPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::DefaultedDestructuringAssignment(
+//             ap,
+//         ))
+//     }
+// }
```

Comment:
Probably don't check in all the commented-out code?

Reply:
Fixed, thanks!